opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzersInfo.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opengrok.indexer.analysis;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+/**
+ * A class wrapping information about used analyzers.
+ */
+public class AnalyzersInfo {
+
+    /**
+     * Descending string length comparator for magics
+     */
+    private static final Comparator<String> DESCENDING_LENGTH_COMPARATOR =
+            Comparator.comparingInt(String::length).thenComparing(String::toString);
+
+    /**
+     * Modified when
+     * {@link AnalyzerFramework#addExtension(String, AnalyzerFactory)}
+     * or
+     * {@link AnalyzerFramework#addPrefix(String, AnalyzerFactory)}
+     * are called to augment the value in {@link AnalyzerGuru#getVersionNo()}.
+     */
+    public final SortedSet<String> customizations;
```

Comment:
Could you please add getters? It just does't feel right this way.

Reply:
I intentionally don’t want to use getters and setters as this class does no provide any functionality

Reply:
Even simple DTO classes are always written with getters/setters.

Reply:
I’ve seen codes which don’t follow that. If this is code style for opengrok I can change it of course.